light-client/Cargo.toml

```diff
@@ -31,7 +31,7 @@ prost-amino = "0.6.0"
 serde = "1.0.106"
 serde_cbor = "0.11.1"
 serde_derive = "1.0.106"
-sled = "0.33.0"
+sled = "0.34.3"
```

Comment:
I see in the Light Client we've pinned the patch version numbers of our dependencies. Should we be? See [this comment](https://github.com/informalsystems/tendermint-rs/pull/516#discussion_r469327310). (This is more of a general question than specific to this PR).

Does `sled` release breaking changes with patch version increments? Doesn't seem like it to me.

Reply:
It's not really pinning to a specific version, it's equivalent to "^0.34.3" which means >=0.34.3, <0.35.0. I agree that we should handle this consistently across crates though.

Reply:
I agree too and at first I wanted to just use "0.34". I kept this to keep it consistent with all other dependencies in the file.

I think opening a PR to fixing this in all `Cargo.toml`s in the repo makes more sense.